### PR TITLE
XGBoost 2 features samples

### PR DESCRIPTION
**Pull request by raoberman**
# Adding New Samples

## Description

Added 2 new samples to XGBoost including README and sample.json

## Checklist
Administrative
- [ ] Review sample design with the appropriate [Domain Expert](https://github.com/oneapi-src/oneAPI-samples/wiki/Reviewers-and-Domain-Experts): <insert Name Here>
- [ ] If you dependencies/binaries, inform the oneAPI Code Samples Project Manager: @JoeOster

Code Development
- [X] Implement coding guidelines and ensure code quality. [see wiki for details](https://github.com/oneapi-src/oneAPI-samples/wiki/General-Code-Guidelines)
- [X] Adhere to readme template 
- [X] Enforce format via clang-format config file
- [X] Adhere to sample.json specification. https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-specification
- [X] Ensure/create CI test configurations for sample (ciTests field) https://github.com/oneapi-src/oneAPI-samples/wiki/sample-json-ci-test-object
- [X] Run jsonlint on sample.json to verify json syntax. www.jsonlint.com

Security and Legal
- [ ] OSPDT Approval (see @JoeOster for assistance)
- [ ] Compile using the following compiler flags and fix any warnings, the flags are: "/Wall -Wformat-security -Werror=format-security"
- [x] Protex (see @JoeOster for assistance)
- [X] Bandit Scans (Python only)
- [X] IPLDT Scan (see @JoeOster for assistance) JEO-5/14
- [ ] Virus scan

Review
- [ ] Review DPC++ code with Paul Peterseon. (GitHub User: pmpeter1)
- [ ] Review readme with Tom Lenth(@tomlenth) and/or Joe Oster(@JoeOster)
- [X] Tested using Dev Cloud when applicable

### CODE REVIEW

**Comment by raoberman** [comment]
@JoeOster please conduct IP scan and anything else @tomlenth @JoeOster please conduct README review

**Review by JoeOster — CHANGES_REQUESTED** [review]
I made some minor edits, please review and alter as needed

File: AI-and-Analytics/Features-and-Functionality/IntelPython_XGBoost_Performance/README.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
+# `Intel Python XGBoost Performance Sample`
+This sample code illustrates how to analyze the performance benefit from using Intel optimizations upstreamed by Intel to latest XGBoost compared to un-optimized XGBoost 0.81. It demonstrates how to use software products that can be found in the [Intel® oneAPI AI Analytics Toolkit](https://software.intel.com/content/www/us/en/develop/tools/oneapi/ai-analytics-toolkit.html). 
+
+| Optimized for                     | Description
+| :---                              | :---
+| OS                                | 64-bit Linux: Ubuntu 18.04 or higher, 64-bit Windows 10, macOS 10.14 or higher
```
[the remark this answers is not in the document]
Comment (raoberman) [reply AI-and-Analytics/Features-and-Functionality/IntelPython_XGBoost_Performance/README.md:6]: the sample can be run for all OS systems, but AI Kit is only available on Linux. hence the discrepancy. I can remove the other 2 OS depending on your judgement.

File: AI-and-Analytics/Features-and-Functionality/IntelPython_XGBoost_daal4pyPrediction/README.md [hunk 2]
```diff
@@ -0,0 +1,105 @@
+# `Intel Python XGBoost daal4py Prediction Sample`
+This sample code illustrates how to analyze the performance benefit of minimal code changes to port pre-trained XGBoost model to daal4py prediction for much faster prediction than XGBoost prediction. It demonstrates how to use software products that can be found in the [Intel® oneAPI AI Analytics Toolkit](https://software.intel.com/content/www/us/en/develop/tools/oneapi/ai-analytics-toolkit.html). 
+
+| Optimized for                     | Description
+| :---                              | :---
+| OS                                | 64-bit Linux: Ubuntu 18.04 or higher, 64-bit Windows 10, macOS 10.14 or higher
```
[the remark this answers is not in the document]
Comment (raoberman) [reply AI-and-Analytics/Features-and-Functionality/IntelPython_XGBoost_daal4pyPrediction/README.md:6]: the sample can be run for all OS systems, but AI Kit is only available on Linux. hence the discrepancy. I can remove the other 2 OS depending on your judgement.

[the remark this answers is not in the document]
Comment (JoeOster) [reply AI-and-Analytics/Features-and-Functionality/IntelPython_XGBoost_daal4pyPrediction/README.md:6]: Yes please remove, unless your going to add install instructions for Darwin and Windows

File: AI-and-Analytics/Features-and-Functionality/IntelPython_XGBoost_Performance/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (JoeOster) [reply AI-and-Analytics/Features-and-Functionality/IntelPython_XGBoost_Performance/README.md:6]: Yes please remove, unless your going to add install instructions for Darwin and Windows

**Comment by JoeOster** [comment]
As you have this marked as pytrhon, I will need a bandit scan of the python files

**Comment by raoberman** [comment]
> As you have this marked as pytrhon, I will need a bandit scan of the python files

I already ran Bandit scans on these prior to submitting PR. Both came up with 0 issues or vulnerabilities.

**Review by tomlenth — COMMENTED** [review]
Can this sample be run in the devlcoud?  If so, it probably needs the content specific to selecting a node.  Running Samples In DevCloud
If running a sample in the Intel DevCloud, remember that you must specify the compute node (CPU, GPU, FPGA) and whether to run in batch or interactive mode. For more information, see the Intel® oneAPI Base Toolkit Get Started Guide (https://devcloud.intel.com/oneapi/get-started/base-toolkit/)

**Comment by raoberman** [comment]
@tomlenth this sample can be run on DevCloud - I used ai kit JupyterLab which by default is CPU node

File: AI-and-Analytics/Features-and-Functionality/IntelPython_XGBoost_daal4pyPrediction/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (raoberman) [reply AI-and-Analytics/Features-and-Functionality/IntelPython_XGBoost_daal4pyPrediction/README.md]: done

**Review by JoeOster — APPROVED** [review]
approved for ci to run

**Comment by raoberman** [comment]
@JoeOster try again to run CI. Is runipy installed by default as part of base env ai kit? I recall some conversation around that for sample CI issues like this....

**Review by JoeOster — APPROVED** [review]
approved for ci only

**Comment by raoberman** [comment]
@JoeOster fixed the strange errors. approve to run CI again please

**Review by JoeOster — APPROVED** [review]
approved for ci only

**Review by JoeOster — DISMISSED** [review]
approved for ci only testing

**Comment by raoberman** [comment]
@JoeOster please approve. trying a new ci sample.json method.

**Comment by raoberman** [comment]
@mdbtucker  please approve for CI run

**Review by JoeOster — DISMISSED** [review]
approved for ci

**Review by JoeOster — DISMISSED** [review]
approved for ci

**Comment by raoberman** [comment]
@JoeOster @mdbtucker please approve for ci run (this may be it)!

**Review by JoeOster — APPROVED** [review]
approved for ci

**Pull request merged**